$OpenBSD: patch-config-scripts_cups-compiler_m4,v 1.8 2013/10/27 08:50:17 ajacoutot Exp $

XXX push upstream
Don't only test wether GCC support -pie but also test wether a program
built with -pie actually runs; this fixes the build on arm.

--- config-scripts/cups-compiler.m4.orig	Tue Jul 23 14:21:51 2013
+++ config-scripts/cups-compiler.m4	Sun Oct 27 09:04:10 2013
@@ -143,7 +143,7 @@ if test -n "$GCC"; then
 
 			*)
 				CFLAGS="$CFLAGS -fPIE -pie"
-				AC_TRY_COMPILE(,,[
+				AC_RUN_IFELSE(AC_LANG_PROGRAM(),[
 					PIEFLAGS="-fPIE -pie"
 					AC_MSG_RESULT(yes)],
 					AC_MSG_RESULT(no))
